Update pop readme rst

johnsonb-ucar opened the pull request:
- `README.rst` is updated for consistency with the Google developers style guide
- `CESM_DART_config` is renamed `CESM_DART_config.csh` to make it clear it is a cshell script
- `CESM2_1_setup_hybrid` is renamed `setup_CESM_hybrid_ensemble.csh` and `CESM2_1_setup_pmo` is renamed `setup_CESM_perfect_model.csh` to make it clear they are cshell scripts and to remove the CESM release version from the filenames. Removing the release version from the filenames simplifies the documentation. 
- `copy_POP_JRA_restarts.py` is updated to name the directory into which restart files are copied as `stagedir` in order to make the variable name consistent with `DART_params.csh`

While all commits are important in their own way, the most important commit here is the rewrite of `README.rst`.

hkershaw-brown reviewed (changes requested):
This review is coming two parts. Part 1: script changes

Adding .csh to indicate a shell script is sensible.  The downside is if any other scripts are expecting the old name. 

For copy_POP_JRA_restarts.py,  I’ve split this into definitely change, good idea to change.

#### Definitely change:
Casename = g210.G_JRA.v14.gx1v7.01
is outside 
“#    First load from the directory of MONTHLY saves    “ section

then
Casename = g210.G_JRA.v14.gx1v7.01.dr
is inside 
“#         Then load the rest from DAILY saves” section

I suggest moving the first instance  of casename inside the  monthly saves section so it is a bit clearer that the scope of first instance of case name is for the monthly section. 

#### Good idea to change 

I would suggest setting  case_name  once.   Reasoning: I think having the case name declared  a couple of times is prone to error.

At the moment there is:
case_name = g210.G_JRA.v14.gx1v7.01
case_name = g210.G_JRA.v14.gx1v7.01.dr
case_name_out = g210.G_JRA.v14.gx1v7.01

hkershaw-brown commented on models/POP/README.rst:
```diff
@@ -6,232 +6,344 @@ Contents
-tested under the Lanai framework, only use with **CESM** is supported under the 
-Manhattan framework.
+This document describes the DART interface to the Parallel Ocean Program (POP).
+It covers the `Development history`_ of the interface with two implementations
```
````suggestion
It covers the `development history`_ of the interface with two implementations
````

johnsonb-ucar replied:
I gleaned the link text capitalization rules from the [Link text](https://developers.google.com/style/link-text) section of the Google developer style guide. In brief:

1. Section headers use sentence capitalization
2. Mid-sentence link text copies section headers

The end result is that there's a capitalization mid-sentence, which seems a little strange. But the style guide claims that this linking structure minimizes link fatigue. I'm not tied to this structure so if we don't want to follow it, that's fine by me. I just wanted to explain why I wrote links this way.

hkershaw-brown replied:
oh I totally read the link text section as  "capialize as part of a sentence". 
I'll get rid of the suggestions.

hkershaw-brown commented on models/POP/README.rst:
````suggestion
.. _development history
Development History
````

hkershaw-brown commented on models/POP/README.rst:
```diff
@@ -6,232 +6,344 @@ Contents
+
+This document also provides `Detailed instructions for using DART and CESM POP2
+on NCAR's supercomputer`_, including information about the availability of
+restart files for `Creating an initial ensemble`_ of model states and
```
````suggestion
restart files for `creating an initial ensemble`_ of model states and
````

hkershaw-brown commented on models/POP/README.rst:
````suggestion
.. _creating an initial ensemble
Creating an initial ensemble
````

hkershaw-brown commented on models/POP/README.rst:
```diff
@@ -6,232 +6,344 @@ Contents
+This document also provides `Detailed instructions for using DART and CESM POP2
+on NCAR's supercomputer`_, including information about the availability of
+restart files for `Creating an initial ensemble`_ of model states and
+`Observation sequence files`_ for assimilation.
```
````suggestion
`observation sequence files`_ for assimilation.
````

hkershaw-brown commented on models/POP/README.rst:
```diff
@@ -6,232 +6,344 @@ Contents
+- the Community Earth System Model Parallel Ocean Program 2
+  (CESM POP2; Smith et al. 2010 [1]_).
+
+This document also provides `Detailed instructions for using DART and CESM POP2
```
````suggestion
This document also provides `detailed instructions for using DART and CESM POP2
````

hkershaw-brown commented on models/POP/README.rst:
````suggestion
.. _detailed instructions for using DART and CESM POP2 on NCAR's supercomputer
Detailed instructions for using DART and CESM POP2 on NCAR's supercomputer
````

hkershaw-brown commented on models/POP/README.rst:
````suggestion
.. _observation sequence files
Observation sequence files
````

hkershaw-brown commented:
part 2a. 
I really like the flow with the links to documentation sections as part of the sentence. I think the capitalization is a bit jarring so I made lower case links. I'm not super sure of the rst syntax (.. _development history) so the changes are a batch of suggestions so you can accept them all at once.

hkershaw-brown commented:
Part 2b. (last part)
Nice description of the scripts and staging the restarts. It seems like a great improvement.  I think the "Detailed description" section would benefit from a summary of the steps at the beginning, e.g. 

1. Edit dart_params.csh for your specific experiment.
3. Run the appropriate DART setup scripts to create and build the CESM case.
2. Stage your initial ensemble using copy_POP_JRA_restarts.py.
4. The DART setup script will print instructions to the screen.  Follow these instructions to submit your case.

johnsonb-ucar commented:
> This review is coming two parts. Part 1: script changes
> 
> Adding .csh to indicate a shell script is sensible. The downside is if any other scripts are expecting the old name.

Before making the pull request, I edited all of the requisite scripts in the directory to ensure that they expect cshell scripts to have the csh suffix.

> 
> For copy_POP_JRA_restarts.py, I’ve split this into definitely change, good idea to change.
> 
> #### Definitely change:
> Casename = g210.G_JRA.v14.gx1v7.01
> is outside
> “# First load from the directory of MONTHLY saves “ section
> 
> then
> Casename = g210.G_JRA.v14.gx1v7.01.dr
> is inside
> “# Then load the rest from DAILY saves” section
> 
> I suggest moving the first instance of casename inside the monthly saves section so it is a bit clearer that the scope of first instance of case name is for the monthly section.
> 
> #### Good idea to change
> I would suggest setting case_name once. Reasoning: I think having the case name declared a couple of times is prone to error.
> 
> At the moment there is:
> case_name = g210.G_JRA.v14.gx1v7.01
> case_name = g210.G_JRA.v14.gx1v7.01.dr
> case_name_out = g210.G_JRA.v14.gx1v7.01

I resolved these suggestions in the update to `copy_POP_JRA_restarts.py` in commit 93ec11a.

- the `case_name` assignment occurs after the banner comment describing the code block.
- `case_name` is only assigned once.
- `case_name_out` is no longer assigned.
-  `case_name_dr = g210.G_JRA.v14.gx1v7.01.dr` replaces the now-eliminated second assignment of `case_name`.

After making the edits to `copy_POP_JRA_restarts.py`, I tested the script to ensure it runs properly. The restart files were correctly copied to this directory:

`/glade/scratch/johnsonb/g210.G_JRA.v14.gx1v7.01/rest/2010-01-01-00000`

> I think the "Detailed description" section would benefit from a summary of the steps at the beginning, e.g.
> 1. Edit dart_params.csh for your specific experiment.
> 2. Run the appropriate DART setup scripts to create and build the CESM case.
> 3. Stage your initial ensemble using copy_POP_JRA_restarts.py.
> 4. The DART setup script will print instructions to the screen. Follow these instructions to submit your case.

I added these summary steps in commit f4afa28.

hkershaw-brown reviewed (approved):
This looks great. Thanks Ben!
